service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/AllocationDecisionControllerAboutToSubmitTest.java

```diff
@@ -0,0 +1,111 @@
+package uk.gov.hmcts.reform.fpl.controllers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableMap;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.OverrideAutoConfiguration;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.model.AllocationDecision;
+import uk.gov.hmcts.reform.fpl.model.AllocationProposal;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@ActiveProfiles("integration-test")
+@WebMvcTest(AllocationDecisionController.class)
+@OverrideAutoConfiguration(enabled = true)
+class AllocationDecisionControllerAboutToSubmitTest {
+
+    private static final String AUTH_TOKEN = "Bearer token";
+
+    @Autowired
+    private  ObjectMapper mapper;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Test
+    void shouldAddCorrectAllocationDecisionProposal() throws Exception {
+
+        AllocationDecision allocationDecisionWithoutProposal = AllocationDecision.builder()
```

Comment:
This allocationDecision creation is a bit duplication, can you extract to the method please?

Reply:
done